source/beamline/AMBeamline.cpp

```diff
@@ -192,3 +194,29 @@ void AMBeamline::initializeBeamlineSupport(){
 	AMBeamlineSupport::setBeamlineDetectorSetAPI(AMBeamline::bl());
 	AMBeamlineSupport::setBeamlineSynchronizedDwellTimeAPI(AMBeamline::bl());
 }
+
+AMAction3* AMBeamline::createScanInitializationAction(AMScanConfiguration *configuration)
+{
+	AMAction3* result = 0;
+
+	AMStepScanConfiguration *stepScanConfiguration = qobject_cast<AMStepScanConfiguration*>(configuration);
+
+	if (stepScanConfiguration) {
+
+		AMListAction3 *initializationAction = new AMListAction3(new AMListActionInfo3("Initialize step scan axis controls", "Initialize step scan axis controls"), AMListAction3::Parallel);
```

Comment:
I was thinking how about creating a function, maybe called `createInitializingAxisControlsAction()`, for this? so that we can call this function in createScanInitializationAction() of BioXASBeamline and ReixsBeamline?